src/inmanta/moduletool.py

```diff
@@ -290,6 +304,91 @@ def install(self) -> None:
         project: Project = self.get_project(load=False)
         project.install_modules()
 
+    def update(self, module: Optional[str] = None, project: Optional[Project] = None) -> None:
+        """
+        Update all modules to the latest version compatible with the given module version constraints.
+        """
+
+        if project is None:
+            # rename var to make mypy happy
+            my_project = self.get_project(load=False)
+        else:
+            my_project = project
+
+        def do_update(specs: "Dict[str, List[InmantaModuleRequirement]]", modules: List[str]) -> None:
+            v2_modules = {module for module in modules if my_project.module_source.path_for(module) is not None}
+
+            v2_python_specs: List[Requirement] = [
+                ModuleV2Source.get_python_package_requirement(module_spec)
+                for module, module_specs in specs.items()
+                for module_spec in module_specs
+                if module in v2_modules
+            ]
+            if v2_python_specs:
+                env.process_env.install_from_index(
+                    v2_python_specs,
+                    my_project.module_source.urls,
+                    upgrade=True,
+                    allow_pre_releases=my_project.install_mode != InstallMode.release,
+                )
+
+            for v1_module in set(modules).difference(v2_modules):
+                spec = specs.get(v1_module, [])
+                try:
+                    ModuleV1.update(my_project, v1_module, spec, install_mode=my_project.install_mode)
+                except Exception:
+                    LOGGER.exception("Failed to update module %s", v1_module)
+
+            # Load the newly installed modules into the modules cache
+            my_project.install_modules(bypass_module_cache=True)
```

Comment:
shouldn't this set 'update_dependencies' ?

Reply:
Wow, I've missed some quite important things. Thanks.

Reply:
Well, they nicely cancel out